aws/dynamodb/dynamo.tf

```
@@ -2,7 +2,7 @@ resource "aws_dynamodb_table" "reliability_queue" {
   name                        = "ReliabilityQueue"
   billing_mode                = "PAY_PER_REQUEST"
   hash_key                    = "SubmissionID"
-  deletion_protection_enabled = true
+  deletion_protection_enabled = var.env == "development" ? false : true
```

A minor tweak if you are interested in simplifying the expressions:
```suggestion
  deletion_protection_enabled = var.env != "development"
```